game-core/src/main/java/games/strategy/engine/framework/ArgParser.java

```diff
@@ -35,7 +35,7 @@ public ArgParser(final Set<String> availableProperties) {
    * @return Return true if all args were valid and accepted, false otherwise.
    */
   public boolean handleCommandLineArgs(final String[] args) {
-    resetTransientClientSettings();
+    ClientSetting.MAP_FOLDER_OVERRIDE.save(ClientSetting.MAP_FOLDER_OVERRIDE.defaultValue);
```

Comment:
A little history here...  The reason for the extracted method with the detailed comment was to avoid someone in the future wondering why this one (or possibly more eventually) property was being reset.  Without some context, they may simply remove it or move it around in this method, which may have an undesirable side effect.

Granted, there is a unit test to cover this case, but the statement on its own is definitely a WTF moment if you don't know why it's there.

#2321 has all the details.

Reply:
I did not find the comment to be very valuable, it did not tell me why of much, just what, which can be inferred from the code directly. Is there a real warning we should have here? We've got a number of places where we have pretty mysterious calls, like the clearing of AI cache; I'm curious if this case is that much different.